### PR TITLE
admission: fold create+update validation funcs into update ObjectInfo

### PR DESCRIPTION
**Pull request by sttts**
Follow-up of #54484.

Pure simplification of the func signatures. We expanded them with two admission validation closures. This PR takes that back partly, addressing what @lavalamp suggested https://github.com/kubernetes/kubernetes/pull/54484#discussion_r147200200. There is no functional change here at all.

### CODE REVIEW

**Comment by caesarxuchao** [comment]
@sttts could you provide more description for the PR? I didn't follow last PR closely so I don't see the purpose of the change.

**Comment by sttts** [comment]
> @sttts could you provide more description for the PR? I didn't follow last PR closely so I don't see the purpose of the change.

Pure simplification of the func signatures. We expanded them with two admission validation closures. This PR takes that back partly, addressing what @lavalamp suggested https://github.com/kubernetes/kubernetes/pull/54484#discussion_r147200200. There is no functional change here at all.

File: staging/src/k8s.io/apiserver/pkg/registry/rest/rest.go [hunk 1]
```diff
@@ -219,6 +219,12 @@ type UpdatedObjectInfo interface {
 	// UpdatedObject returns the updated object, given a context and old object.
 	// The only time an empty oldObj should be passed in is if a "create on update" is occurring (there is no oldObj).
 	UpdatedObject(ctx genericapirequest.Context, oldObj runtime.Object) (newObj runtime.Object, err error)
+
+	// ValidateCreate applies validating admission to a newly created object.
+	ValidateCreate(obj runtime.Object) error
```
Comment (deads2k) [inline staging/src/k8s.io/apiserver/pkg/registry/rest/rest.go]: @liggitt you're the original author.  Yay or nay?  I don't feel strongly one way or the other.

Comment (sttts) [reply staging/src/k8s.io/apiserver/pkg/registry/rest/rest.go]: The name `ObjectInfo` does not match anymore with this change. It's more an `UpdateInfo` now.

**Comment by sttts** [comment]
@liggitt this needs your opinion: bigger ObjectInfo or longer function signatures?

**Comment by liggitt** [comment]
UpdatedInfo was all about encapsulating oldObject->newObject. At first glance, validation doesn't seem to be related to that.

I didn't follow the admission rewire closely... does the rest handler still have the ability to modify the object that is returned from UpdatedObjectInfo before persisting it?

**Comment by sttts** [comment]
> UpdatedInfo was all about encapsulating oldObject->newObject. At first glance, validation doesn't seem to be related to that.

That's why I would call it UpdateInfo, i.e. everything that specifies the update. Then validation is related.

> does the rest handler still have the ability to modify the object that is returned from UpdatedObjectInfo before persisting it?

Not sure I can follow. If you can link some code where this is done.... I don't think we have changed anything in this direction.

**Comment by techtonik** [comment]
Still needs rebase with conflict resolution.

**Pull request closed without merging**